scrapy/link.py

```diff
@@ -28,6 +28,7 @@ def __init__(self, url, text='', fragment='', nofollow=False):
         self.text = text
         self.fragment = fragment
         self.nofollow = nofollow
+        self.encoding = encoding
 
     def __eq__(self, other):
         return self.url == other.url and self.text == other.text and \
```

Comment:
I'm worried about performance because link extractors are often on a hot path: pages may have hundreds of links